packages/niivue/src/nvimage/index.ts

```diff
@@ -3352,6 +3378,42 @@ export class NVImage {
         dataBuffer = null
       }
     }
+    const re = /(?:\.([^.]+))?$/
+    let ext = ''
+    // if zarr
+    ext = re.exec(url)[1]
+    // try url and name attributes to test for .zarr
+    if (ext === 'zarr' || re.exec(name)[1] === 'zarr') {
+      const store = new zarr.FetchStore(url)
+      const arr = await zarr.open(store, { kind: 'array' })
```

Comment:
NB: zarrita has some [type-level helpers](https://zarrita.dev/packages/zarrita.html#data-typing-in-typescript) that I sprinkle in often with some assertions to make some high-level checks about the data source. For instance, in the code below, object, string, or bigint arrays would throw an error.

```ts
arr // zarr.Array<zarr.DataType, zarr.FetchStore)>
if (!arr.is("number")) {
  throw new Error(`Unsupported zarr data_type, got: ${arr.dtype}.`)
}
arr // zarr.Array<"int8" | "int16" | "int32" | "uint8" | "uint16" | "uint32" | "float32"| "float64", zarr.FetchStore>
```

The `.is` method hooks into TypeScript's type narrowing, and the resulting `chunk.data` from `zarr.get` will be the appropriate union type of TypedArrays.